### ISSUE TITLE
Image Size in the dataset

### ISSUE THREAD

**Issue by waqarsqureshi** [open-mmlab/mmsegmentation#1577]
@[MengzhangLI](https://github.com/open-mmlab/mmsegmentation/commits?author=MengzhangLI) Please correct If I am wrong.

While I understand is the image size depends on the actual size of the image in the dataset. It can only change randomly in the pipeline along with other augmentations applied during training including cropping. Am I right? this means the cityscape image size will always be 2048 x 1024 in the pipeline. The resizing followed by cropping is just the augmentation done on the input image.

Another thing that confuses me is how you handle resizing the mask. Usually, in opencv, I have seen that using nearest-neighbor interpolation on the edges of two classes. It uses the middle of the two values. I see you use resize function defined in mmcv and using the interpolation nearest. Is it different than OpenCV or how does it handle corner cases?

**Comment by MengzhangLI** [comment]
Hi, @waqarsqureshi 

Yes, you are right. In training, we usually use `RandomCrop` to crop part of resizing images into a fixed shape for next transformation in pipeline. In inference, whole image is feed to network so resizing or cropping are unnecessary.

Resize function is adopted in MMSegmentation [here](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/ops/wrappers.py#L8-L27), the key operation is:

```python
F.interpolate(input, size, scale_factor, mode, align_corners)
```

However, we do not make specifical comparison with OpenCV.

**Comment by waqarsqureshi** [comment]
Does this mean that a random crop of each image to the size given in crop_size(x,x)? i.e. the location of the crop is random, while every image is cropped in the pipeline after resizing in the training pipeline. The final image fed to the model should be the same size either the original size or the cropped image. 
I think there is a problem in assigning labels to each image before interpolation. the nearest neighbor is working fine. May be it is the border pixel in the label-bee client. I am not sure yet.

**Comment by MengzhangLI** [comment]
> Does this mean that a random crop of each image to the size given in crop_size(x,x)? i.e. the location of the crop is random, while every image is cropped in the pipeline after resizing in the training pipeline. The final image fed to the model should be the same size either the original size or the cropped image. I think there is a problem in assigning labels to each image before interpolation. the nearest neighbor is working fine. May be it is the border pixel in the label-bee client. I am not sure yet.

(1) Yes, Waqar you are right.

(2) We did not conduct experiments on different interpolation modes, but based on previous re-impelmenting many models, it would not fluctuate widely. You could have a try and looking forward to your reply!

Best,

**Comment by waqarsqureshi** [comment]
Thanks the issue in interpolation was not due to OpenCV nearest neighbour. It was from the labeled-client export function. I have raised an Issue there. Thank you.
